[HttpClient] Symfony HTTP Client documentation (#19170)
symfony/symfony-docs: http_client.rst

Chunked transfer encoding is enabled automatically if both your PHP runtime and

> the remote server supports it.

Reviewer: ```suggestion
the remote server support it.
```

.. caution::

>     Symfony automatically adds the `Accept-Encoding: gzip` HTTP header if certain requirements are met. However, be aware that if you manually set `Accept-Encoding` to `gzip`, you will need to handle the decompression yourself.

Reviewer: ```suggestion
    If you set `Accept-Encoding` to e.g. `gzip`, you will need to handle the decompression yourself.
```
